tartiflette/engine.py

```diff
@@ -96,79 +96,53 @@ def __init__(
-
-                engine = await create_engine(
-                    sdl,
-                    schema_name = "default",
-                    error_coercer = None,
-                    custom_default_resolver = None,
-                    modules = None,
-                )
-                ```
-
-                More details on the website: https://tartiflette.io/docs/api/engine#create_engine-prepares-and-cooks-your-engine
-                """
-            )
+        self._schema_name = schema_name
+        self._error_coercer = error_coercer
+        self._custome_default_resolver = custom_default_resolver
+        self._modules = modules
+        self._sdl = sdl
 
     async def cook(
         self,
-        sdl: Union[str, List[str]],
+        sdl: Union[str, List[str]] = None,
         error_coercer: Callable[[Exception], dict] = None,
         custom_default_resolver: Optional[Callable] = None,
         modules: Optional[Union[str, List[str]]] = None,
-        schema_name: str = "default",
+        schema_name: str = None,
     ):
         """
         Cook the tartiflette, basicly prepare the engine by binding it to given modules using the schema_name as a key.
         You wont be able to execute a request if the engine wasn't cooked.
 
-        Arguments:
-            sdl {Union[str, List[str]]} -- The SDL to work with.
-
         Keyword Arguments:
+            sdl {Union[str, List[str]]} -- The SDL to work with.
             schema_name {str} -- The name of the SDL (default: {"default"})
             error_coercer {Callable[[Exception, dict], dict]} -- An optional callable in charge of transforming a couple Exception/error into an error dict (default: {default_error_coercer})
             custom_default_resolver {Optional[Callable]} -- An optional callable that will replace the tartiflette default_resolver (Will be called like a resolver for each UNDECORATED field) (default: {None})
             modules {Optional[Union[str, List[str]]]} -- An optional list of string containing the name of the modules you want the engine to import, usually this modules contains your Resolvers, Directives, Scalar or Subscription code (default: {None})
         """
 
         if not modules:
-            modules = []
+            modules = self._modules or []
 
         if isinstance(modules, str):
             modules = [modules]
 
+        schema_name = schema_name or self._schema_name or "default"
+
         self._error_coercer = error_coercer_factory(
-            error_coercer or default_error_coercer
+            error_coercer or self._error_coercer or default_error_coercer
         )
+
         self._modules, modules_sdl = await _import_modules(
             modules, schema_name
         )
-        SchemaRegistry.register_sdl(schema_name, sdl, modules_sdl)
+
+        SchemaRegistry.register_sdl(schema_name, sdl or self._sdl, modules_sdl)
         self._schema = SchemaBakery.bake(schema_name, custom_default_resolver)
```

Comment:
Shouldn't it be:
```suggestion
        self._schema = SchemaBakery.bake(schema_name, custom_default_resolver or self. _custome_default_resolver)
```

Also, shouldn't we add a check on the `custom_default_resolver` like we do on `@Resolver`. I mean, we should check that its a coroutine function, isn't?